docs/extensions/autosummary_skip_inherited.py

```diff
@@ -0,0 +1,57 @@
+"""Extension to skip inherited methods and properties in autosummary."""
+
+from __future__ import annotations
+
+from traceback import walk_stack
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Literal
+
+    from sphinx.application import Sphinx
+    from sphinx.ext.autodoc import Options
+
+
+def skip_inherited(  # noqa: PLR0917
+    app: Sphinx,
+    what: Literal[
+        "module", "class", "exception", "function", "method", "attribute", "property"
+    ],
+    name: str,
+    obj: object,
+    skip: bool,  # noqa: FBT001
+    options: Options | dict[str, object],
+) -> bool | None:
+    """Skip inherited members."""
+    # Skip `getdoc` property
+    if what == "method" and name == "getdoc":
+        return True
+
+    # find parent class
+    for frame, _ in walk_stack(None):
+        # Sadly `autodoc-skip-member` doesn’t give access to the parent object, so we need to do this.
+        # Find this stack frame: https://github.com/sphinx-doc/sphinx/blob/a5366394ae527712c4edfeb07a5fbeecd4ca72e1/sphinx/ext/autosummary/generate.py#L496-L517
+        if frame.f_code.co_name == "_get_members" and frame.f_code.co_filename.endswith(
+            "/generate.py"
+        ):
+            parent = frame.f_locals["obj"]
+            if not isinstance(parent, type):
+                return None
+            break
+    else:
+        return None
+
+    # return if it’s a member of the parent class
+    typ = parent
+    while typ is not type:
+        if name in typ.__dict__:
+            return None
+        typ = type(typ)
```

Comment:
Is the `while` loop really necessary? I would think if `parent` is something like
```python
class B:
    pass
parent = B
```
then we can just do a simply `if name in parent.__dict__` check.

So this might look something like

```python
if type(parent) is type and if name not in parent.__dict__:
    return True
return None
```
i.e., the `parent`'s type is `type` (so presumably it has members) and `name` is not a member.  Otherwise fall back.

Perhaps my understanding is flawed

Reply:
yes it is, because of metaclasses. it was like you said, but didn’t work for the metaclass properties.

Reply:
I had considered that, but didn't want to look foolish suggesting it and then being wrong.  Oh well.

```suggestion
        # Metaclasses need the while loop to trace back to their parents.
        # See `SingletonMeta` and the documentation of `settings` for an example.
        typ = type(typ)
```

Reply:
yeah, the only way `type(thing) is type` doesn’t apply is with metaclasses.

I guess I could have unrolled this to only allow one level of metaclass, but why not like this?

Reply:
I'm fine with it the way it is.   Just wanted to be clear about what was going on.